doc/workshops/plugin-development.rst

```diff
@@ -2,4 +2,917 @@
+
+        # --- SpyderPluginV2 API
+        # ------------------------------------------------------------------------
+        def get_name(self):
+            return _("Spyder Pomodoro Timer")
+
+        def get_description(self):
+            return _("A very simple pomodoro timer")
+
+        def get_icon(self):
+            return QIcon()
+
+        def register(self):
+            container = self.get_container()
+            print('SpyderPomodoroTimer registered!')
+
+        def check_compatibility(self):
+            valid = True
+            message = ""  # Note: Remember to use _("") to localize the string
+            return valid, message
+
+        def on_close(self, cancellable=True):
+            return True
+
+For now, we only need to define the dependencies of our plugin, redefining ``REQUIRES``.
+
+.. code-block:: python
+
+       REQUIRES = [Plugins.StatusBar]
+
+We can follow a similar approach for the icon, substituting,
+
+.. code-block:: python
+
+   from qtpy.QtGui import QIcon
+
+   # ...
+
+.. code-block:: python
+
+       def get_icon(self):
+           return QIcon()
+
+by
+
+.. code-block:: python
+
+   import qtawesome as qta
+
+   # ...
+
+.. code-block:: python
+
+       def get_icon(self):
+           return qta.icon("mdi.av-timer", color=ima.MAIN_FG_COLOR)
+
+and change,
+
+.. code-block:: python
+
+       def register(self):
+           container = self.get_container()
+           print('SpyderPomodoroTimer registered!')
+
+to
+
+.. code-block:: python
+
+       def register(self):
+           container = self.get_container()
+           statusbar = self.get_plugin(Plugins.StatusBar)
+           if statusbar:
+               statusbar.add_status_widget(self.pomodoro_timer_status)
+           print('SpyderPomodoroTimer registered!')
+
+So, basically, Spyder is aware of the presence of our plugin, and that this plugin adds a new widget to the status bar.
+
+At the end of ``plugin.py`` we add the following:
+
+.. code-block:: python
+
+   @property
+   def pomodoro_timer_status(self):
+       container = self.get_container()
+       return container.pomodoro_timer_status
+
+In this way, ``SpyderPomodoroTimer`` can access ``pomodoro_timer_status`` of ``SpyderPomodoroTimerContainer`` as if it were its own property.
+
+In summary, we have been doing the following:
+
+.. image:: images/pd_plugin_structure.png
+   :alt: Basic structure of Pomodoro Timer Spyder plugin.
+
+We have created a widget, then we add the widget to the container, from there we register the container in the plugin next to the status bar. In the plugin, we accessed the widget and inserted it into the status bar.
+
+The use of the container is necessary because Qt requires our widget to be inside another widget before using it (otherwise it would become a floating window). And through the container that is derived from ``SpyderWidgetMixin`` and ``SpyderToolbarMixin`` mixins it gets integrated to the rest of Spyder.
+
+======================
```

Comment:
I would add a link on top of this to the "solutions" code for people to see how their code should be looking like

Reply:
Done, I added the link to the repository.